src/drivers/intel/cavs/alh.c

```diff
@@ -29,21 +30,37 @@ static int alh_trigger(struct dai *dai, int cmd, int direction)
 
 static int alh_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 {
+	struct alh_pdata *alh = dai_get_drvdata(dai);
+
 	dai_info(dai, "alh_set_config() config->format = 0x%4x",
 		  config->format);
 
+	alh->params = config->alh;
+
 	return 0;
 }
 
 /* get ALH hw params */
 static int alh_get_hw_params(struct dai *dai,
-			     struct sof_ipc_stream_params  *params, int dir)
+			     struct sof_ipc_stream_params *params, int dir)
 {
+	struct alh_pdata *alh = dai_get_drvdata(dai);
+
+	/*
+	 * ALH config is sent multiple times, only the first batch contains
+	 * the stream format
+	 */
+	if (alh->params.rate || alh->params.channels || !alh->has_params) {
```

Comment:
@sathishkuttan This doesn't work. alh_get_hw_params() will be invoked after two alh_set_config(). 
The flow is:
```
topology load -> alh_set_config() with rate and channels params.
        hw_params -> alh_set_config() with empty params.
                alh_get_hw_params()
```
So alh->params.rate and alh->params.channels will be 0 when alh_get_hw_params() is invoked
So I still think the test should be done in alh_set_config().

Reply:
Oh, yes, you're absolutely right. This is an oversight.